src/Integration/Telemetry/TelemetryManager.cs

```diff
@@ -67,132 +46,116 @@ public bool IsAnonymousDataShared
-                SonarLintVersion = SonarLintVersion,
-                VisualStudioVersion = visualStudioVersion,
-                NumberOfDaysSinceInstallation = numberOfDaysSinceInstallation,
-                NumberOfDaysOfUse = this.telemetryRepository.Data.NumberOfDaysOfUse,
-                IsUsingConnectedMode = this.solutionBindingTracker.IsActiveSolutionBound
-            };
+            telemetryRepository.Data.IsAnonymousDataShared = true;
+            telemetryRepository.Save();
+
+            EnableAllEvents();
         }
 
-        private void EnableAllEvents()
+        public async void OptOut()
         {
-            this.tryUploadDataTimer.Elapsed += OnTryUploadDataTimerElapsed;
-            this.tryUploadDataTimer.Start();
+            telemetryRepository.Data.IsAnonymousDataShared = false;
+            telemetryRepository.Save();
 
-            this.firstCallDelayer.Elapsed += OnTryUploadDataTimerElapsed;
-            this.firstCallDelayer.Start();
+            DisableAllEvents();
 
-            KnownUIContexts.SolutionBuildingContext.UIContextChanged += this.OnAnalysisRun;
-            KnownUIContexts.SolutionExistsAndFullyLoadedContext.UIContextChanged += this.OnAnalysisRun;
+            await telemetryClient.OptOut(GetPayload(telemetryRepository.Data));
         }
 
         private void DisableAllEvents()
         {
-            this.tryUploadDataTimer.Elapsed -= OnTryUploadDataTimerElapsed;
-            this.tryUploadDataTimer.Stop();
+            telemetryTimer.Elapsed -= OnTelemetryTimerElapsed;
+            telemetryTimer.Stop();
 
-            this.firstCallDelayer.Elapsed -= OnTryUploadDataTimerElapsed;
-            this.firstCallDelayer.Stop();
-
-            KnownUIContexts.SolutionBuildingContext.UIContextChanged -= this.OnAnalysisRun;
-            KnownUIContexts.SolutionExistsAndFullyLoadedContext.UIContextChanged -= this.OnAnalysisRun;
+            knownUIContexts.SolutionBuildingContextChanged -= this.OnAnalysisRun;
+            knownUIContexts.SolutionExistsAndFullyLoadedContextChanged -= this.OnAnalysisRun;
         }
 
-        public void OptIn()
+        private void EnableAllEvents()
         {
-            this.telemetryRepository.Data.IsAnonymousDataShared = true;
-            this.telemetryRepository.Save();
+            telemetryTimer.Elapsed += OnTelemetryTimerElapsed;
+            telemetryTimer.Start();
 
-            EnableAllEvents();
+            knownUIContexts.SolutionBuildingContextChanged += this.OnAnalysisRun;
+            knownUIContexts.SolutionExistsAndFullyLoadedContextChanged += this.OnAnalysisRun;
         }
 
-        public async void OptOut()
+        private TelemetryPayload GetPayload(TelemetryData telemetryData)
         {
-            this.telemetryRepository.Data.IsAnonymousDataShared = false;
-            this.telemetryRepository.Save();
+            var numberOfDaysSinceInstallation = (long)DateTime.Now.DaysPassedSince(telemetryData.InstallationDate);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to 'long'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1905)